core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/RequesterPaysSetupServiceSpec.scala

```diff
@@ -129,23 +119,64 @@ class RequesterPaysSetupServiceSpec extends AnyFlatSpec with Matchers with Mocki
 
     // remove users from 1 workspace and check that it did not get removed from google bindings
     service.revokeAllUsersFromWorkspace(minimalTestData.v1Workspace).futureValue should contain theSameElementsAs expectedEmails
-    runAndWait(workspaceRequesterPaysQuery.userExistsInWorkspaceNamespace(minimalTestData.billingProject.projectName.value, userInfo.userEmail)) shouldBe true
-    runAndWait(workspaceRequesterPaysQuery.userExistsInWorkspaceNamespace(minimalTestData.billingProject.projectName.value, minimalTestData.userReader.userEmail)) shouldBe true
     service.googleServicesDAO.asInstanceOf[MockGoogleServicesDAO].policies.get(minimalTestData.v1Workspace.googleProjectId) shouldBe Some(initialBindings)
 
     // remove users from other workspace and check that it did get removed from google bindings
     service.revokeAllUsersFromWorkspace(minimalTestData.v1Workspace2).futureValue should contain theSameElementsAs expectedEmails
-    runAndWait(workspaceRequesterPaysQuery.userExistsInWorkspaceNamespace(minimalTestData.billingProject.projectName.value, userInfo.userEmail)) shouldBe false
-    runAndWait(workspaceRequesterPaysQuery.userExistsInWorkspaceNamespace(minimalTestData.billingProject.projectName.value, minimalTestData.userReader.userEmail)) shouldBe false
+    service.googleServicesDAO.asInstanceOf[MockGoogleServicesDAO].policies.get(minimalTestData.v1Workspace2.googleProjectId) shouldBe Some(Map(service.requesterPaysRole -> Set.empty))
+  }
+
+  "revokeAllUsersFromWorkspace" should "unlink all in namespace if all v1 workspaces are no longer requester pays" in withMinimalTestDatabaseAndServices { service =>
+    val user1SA = BondServiceAccountEmail("bondSA1")
+    val user2SA = BondServiceAccountEmail("bondSA2")
+    val expectedEmails = Set(user1SA, user2SA)
+
+    // add users to 2 workspaces in same namespace
+    runAndWait(workspaceRequesterPaysQuery.insertAllForUser(minimalTestData.v1Workspace.toWorkspaceName, userInfo.userEmail, Set(user1SA)))
+    runAndWait(workspaceRequesterPaysQuery.insertAllForUser(minimalTestData.workspace2.toWorkspaceName, userInfo.userEmail, Set(user1SA)))
+    runAndWait(workspaceRequesterPaysQuery.insertAllForUser(minimalTestData.v1Workspace.toWorkspaceName, minimalTestData.userReader.userEmail, Set(user2SA)))
+    runAndWait(workspaceRequesterPaysQuery.insertAllForUser(minimalTestData.workspace2.toWorkspaceName, minimalTestData.userReader.userEmail, Set(user2SA)))
+
+    // add 2 users to mock google bindings
+    val initialBindings = Map(service.requesterPaysRole -> expectedEmails.map(expectedEmail => "serviceAccount:" + expectedEmail.client_email))
+    service.googleServicesDAO.asInstanceOf[MockGoogleServicesDAO].policies.put(minimalTestData.v1Workspace.googleProjectId, initialBindings)
+
+    // remove users from 1 workspace and check that it did not get removed from google bindings
```

Comment:
This comment seems to be out of date since the check below appears to be asserting that removing users from this workspace does remove the Google bindings